app/lib/user_settings_decorator.rb

```diff
@@ -14,32 +14,35 @@ def update(settings)
 
   private
 
+  # rubocop: disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
```

Comment:
I wouldn't enable that. This class definitely needs refactoring. But that's for another PR (e.g. #18368)

Reply:
A commit that avoids the warning will revert.